Core/Core/Offline/View/UIButtonOfflineSupport.swift

```diff
@@ -0,0 +1,75 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import UIKit
+
+extension UIButton {
+
+    // MARK: - Public Interface
+
+    public func makeUnavailableInOfflineMode(_ interactor: OfflineModeInteractor = OfflineModeInteractorLive.shared) {
+        unowned let uSelf = self
+        let observation = interactor
+            .observeIsOfflineMode()
+            .removeDuplicates()
+            .sink { offlineMode in
+                if offlineMode {
+                    uSelf.makeUnavailable()
+                } else {
+                    uSelf.makeAvailable()
+                }
+            }
+        objc_setAssociatedObject(self, &AssociatedObjectKeys.OfflineStateObservation, observation, .OBJC_ASSOCIATION_RETAIN)
+    }
+
+    // MARK: - Private Methods
+
+    private struct AssociatedObjectKeys {
+        static var OfflineStateObservation = "OfflineStateObservation"
+        static var TapGestureRecognizer = "TapGestureRecognizer"
+    }
+
+    private func makeUnavailable() {
```

Comment:
I'm not sure if this function name is expressive enough. I'm thinking something like `setDisabledState` or `setUnavailableState`.

Reply:
Agree, methods starting with make used to create and return something.